sdp/processors/langs/armenian.py

```diff
@@ -0,0 +1,81 @@
+import os
+import pandas as pd
+from pathlib import Path
+from sdp.processors.base_processor import BaseProcessor, BaseParallelProcessor, DataEntry
+from sdp.utils.common import load_manifest
+
+
+class GetSourceFolder(BaseParallelProcessor):
+    """
+    Processor for extracting source folder from file paths and updating the manifest.
+
+    Args:
+        source_filepath (str): The field containing the file path in the manifest.
+        source_field (str): The field to store the extracted source folder in the manifest.
+        **kwargs: Additional keyword arguments to be passed to the base class `BaseParallelProcessor`.
+
+    Note:
+        This class inherits from the `BaseParallelProcessor` class and extends its functionality to extract source information from file paths and update the manifest.
+    """
+    def __init__(
+        self,
+        source_filepath: str,
+        source_field: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.input_field = source_filepath
+        self.output_field = source_field
```

Comment:
I think it's weird to call the variables one thing in the constructor and then rename them inside the class. Please pick a consistent set of names

Reply:
fixed